packages/dds/tree/src/test/shared-tree/fuzz/fuzzEditGenerators.ts

```diff
@@ -292,174 +299,124 @@ export const makeEditGenerator = (
-				// we're more likely to generate more interesting outcomes.
-				const count = state.random.integer(1, Math.min(3, field.length - start));
-				const node = field.at(start);
-				// We computed 'start' in a way that guarantees it's in-bounds, so at() shouldn't have returned undefined.
-				assert(node !== undefined, "Tried to access a node that doesn't exist");
+			},
+			weights.remove,
+			({ fieldInfo }) => fieldInfo.content.length > 0,
+		],
+		[
+			({ fieldInfo, random }): IntraFieldMove => {
+				const field = fieldInfo.content;
+				const first = random.integer(0, field.length - 1);
+				const last = random.integer(first, field.length - 1);
 				return {
-					type: "sequence",
-					edit: {
-						type: "remove",
-						firstNode: downPathFromNode(node),
-						count,
-					},
+					type: "intraFieldMove",
+					range: { first, last },
+					dstIndex: random.integer(0, field.length),
 				};
-			}
-			default:
-				fail(`Invalid field type for deletion of content: ${fieldInfo.type}`);
-		}
-	};
-
-	const move = (state: FuzzTestState): FieldEditTypes => {
-		const tree = state.client.channel; // TODO: remove if unused
-		const fieldInfo = selectTreeField(
-			viewFromState(state),
-			state.random,
-			weights.fieldSelection,
-			(f) => f.type === "sequence" && f.content.length > 0,
-		);
-		assert(fieldInfo.type === "sequence", "Move should only be performed on sequence fields");
-		const { content: field } = fieldInfo;
-		assert(field.length > 0, "Sequence must have at least one element to perform a move");
-
-		const start = state.random.integer(0, field.length - 1);
-		const count = state.random.integer(1, field.length - start);
-		const dstIndex = state.random.integer(0, field.length);
-		const node = field.at(start);
-		assert(node !== undefined, "Node should be defined at chosen index");
-
-		return {
-			type: "sequence",
-			edit: {
-				type: "move",
-				dstIndex,
-				count,
-				firstNode: downPathFromNode(node),
 			},
-		};
-	};
+			weights.move,
+			({ fieldInfo }) => fieldInfo.content.length > 0,
+		],
+	]);
 
-	const fieldEdit = createWeightedGeneratorWithBailout<FieldEditTypes, FuzzTestState>([
+	const optionalFieldEditGenerator = createWeightedGenerator<
+		OptionalFieldEdit["edit"],
+		FuzzTestStateForFieldEdit<OptionalFuzzField>
+	>([
 		[
-			insert,
-			weights.insert,
-			(state) =>
-				trySelectTreeField(
-					viewFromState(state),
-					state.random,
-					weights.fieldSelection,
-					weights.fieldSelection.filter,
-				) !== "no-valid-fields",
+			(state): SetField => ({
+				type: "set",
+				value: jsonableTree(state),
+			}),
+			weights.set,
+		],
+		[{ type: "clear" }, weights.clear, (state) => state.fieldInfo.content !== undefined],
+	]);
+
+	const requiredFieldEditGenerator = (
+		state: FuzzTestStateForFieldEdit<RequiredFuzzField>,
+	): RequiredFieldEdit["edit"] => ({
+		type: "set",
+		value: jsonableTree(state),
+	});
+
+	const fieldEditGenerator = createWeightedGeneratorWithBailout<
```

Comment:
By the time this generator is invoked, the field is already known. Rather than use `createWeightedGenerator` or some variant, it seems more appropriate to do a switch statement over the field type (I think `state.field.key` works to distinguish the cases?).